static/js/typeahead_helper.js

```diff
@@ -358,14 +359,27 @@ export function sort_slash_commands(matches, query) {
     return results.matches.concat(results.rest);
 }
 
-// Gives stream a score from 0 to 3 based on its activity
+// Gives stream a score from 0 to 5 based on its activity
+// stream that the message is addressed to (if any) > unmuted pinned streams > unmuted subscribed streams > muted streams > other streams
 function activity_score(sub) {
     let stream_score = 0;
     if (!sub.subscribed) {
         stream_score = -1;
     } else {
-        if (sub.pin_to_top) {
+        if (sub.name === compose_state.stream_name()) {
+            stream_score += 4;
+        }
+        if (!sub.is_muted) {
+            stream_score += 3;
+            if (sub.pin_to_top) {
+                stream_score += 2;
```

Comment:
What's the story behind using 1/2/3/4 as the things to add?  Usually if you want to simulate priorities, the things one adds will be powers of 2, e.g. 1/2/4/8.

Reply:
I was trying to follow the previous approach of ranking streams. I will change the 1/2/3/4 to powers of 2 like 1/2/4/8 to simulate priorities.

Reply:
@timabbott I have tweaked ``activity_score`` function in ``typeahead_helper.js`` to prioritize streams properly with powers of 2 eg. 1/2/4/8. Please review my changes and suggest if any changes are required.